packages/test-utils/src/reporter.ts

```diff
@@ -698,7 +700,7 @@ class ReplayReporter {
     recordings.forEach(rec => add(rec.id, mergedMetadata));
 
     // Re-fetch recordings so we have the most recent metadata
-    const allRecordings = listAllRecordings({ all: true });
+    const allRecordings = listAllRecordings({ all: true }) as RecordingEntry<TRecordingMetadata>[];
```

Comment:
This part is still something that I don't quite understand but I take it for granted that it's kinda how it is supposed to be here. Even though the metadata is not owned by `test-utils` - it's currently still initiated here (just a couple of lines above this, using `testMetadata.init(testRun)`, I believe). 

It's written back to disk using `add` and "prefetched" using `listAllRecordings` - so there is no actual full type-safety here hence the usage of the cast.

Reply:
The CLI uses the `recordings.log` as the source of truth for recording status. It doesn't maintain an in-memory view of this data so when we update it on L700 above, we have to re-read it here to get the current data.

Reply:
That I understand - the ownership of this type is confusing to me.